src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,126 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
```

Comment:
Usage of java.util.Objects is not allowed as per the checklist. Consider implementing your own method for object comparison if needed.